File: src/main/java/net/thedragonteam/armorplus/util/TeslaForgeUnitsWrapper.java
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.util;

import net.darkhax.tesla.api.ITeslaConsumer;
import net.darkhax.tesla.api.ITeslaHolder;
import net.darkhax.tesla.api.ITeslaProducer;
import net.minecraftforge.energy.IEnergyStorage;

public class TeslaForgeUnitsWrapper implements ITeslaProducer, ITeslaHolder, ITeslaConsumer {

    private final IEnergyStorage storage;

    public TeslaForgeUnitsWrapper(IEnergyStorage storage) {
        this.storage = storage;
    }

    @Override
    public long givePower(long power, boolean simulated) {
        return this.storage.receiveEnergy((int) power, simulated);
    }

    @Override
    public long getStoredPower() {
        return this.storage.getEnergyStored();
    }

    @Override
    public long getCapacity() {
        return this.storage.getMaxEnergyStored();
    }

    @Override
    public long takePower(long power, boolean simulated) {
        return this.storage.extractEnergy((int) power, simulated);
    }
}

File: src/main/java/net/thedragonteam/armorplus/util/TeslaForgeUnitsWrapper.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.util

import net.darkhax.tesla.api.ITeslaConsumer
import net.darkhax.tesla.api.ITeslaHolder
import net.darkhax.tesla.api.ITeslaProducer
import net.minecraftforge.energy.IEnergyStorage

class TeslaForgeUnitsWrapper(private val storage: IEnergyStorage) : ITeslaProducer, ITeslaHolder, ITeslaConsumer {

    override fun givePower(power: Long, simulated: Boolean): Long {
        return this.storage.receiveEnergy(power.toInt(), simulated).toLong()
    }

    override fun getStoredPower(): Long {
        return this.storage.energyStored.toLong()
    }

    override fun getCapacity(): Long {
        return this.storage.maxEnergyStored.toLong()
    }

    override fun takePower(power: Long, simulated: Boolean): Long {
        return this.storage.extractEnergy(power.toInt(), simulated).toLong()
    }
}